src/goto-programs/goto_convert.cpp

```diff
@@ -1038,9 +1038,11 @@ void goto_convertt::convert_while(
   z->make_skip();
   z->source_location=source_location;
 
+  exprt followed_cond = cond;
+  followed_cond.type() = ns.follow(followed_cond.type());
   goto_programt tmp_branch;
```

Comment:
We are really trying to get rid of these 'follows'. May I suggest you expand while reading in.

Reply:
I'd agree with @kroening - where possible we should expand when reading in the json_symtab, or maybe have a cleanup pass immediately after reading in.